web/src/user_group_pill.ts

```diff
@@ -47,17 +51,18 @@ export function get_group_name_from_item(item: InputPillItem<UserGroupPill>): st
 }
 
 export function get_user_ids(pill_widget: UserGroupPillWidget | CombinedPillContainer): number[] {
-    let user_ids = pill_widget
-        .items()
-        .flatMap((item) =>
-            item.type === "user_group"
-                ? [...user_groups.get_user_group_from_id(item.group_id).members]
-                : [],
-        );
+    let user_ids: number[] = [];
+    for (const user_group_item of pill_widget.items()) {
+        if (user_group_item.type === "user_group") {
+            const user_group = user_groups.get_user_group_from_id(user_group_item.group_id);
+            const group_members = user_groups.get_recursive_group_members(user_group);
+            user_ids.push(...group_members);
+        }
+    }
+
     user_ids = [...new Set(user_ids)];
     user_ids.sort((a, b) => a - b);
 
-    user_ids = user_ids.filter(Boolean);
```

Comment:
This was introduced in ce4cf66f3f80c3b120507d6907b9197c343f81eb, not sure why. Let me know if there's a reason for this to be here and we can drop the first commit that removes this

Reply:
My guess is it was to deal with some previous construction that would put `undefined` values into the list. I think we can remove it, as no such construction is visible here.